luaa.c

```diff
@@ -411,6 +417,94 @@ get_modifier_name(int map_index)
     return 0; /* \0 */
 }
 
+// Helper function for luaA_get_key_name() below.
+// Will return the UTF-32 codepoint IF AND ONLY IF the input is exactly one
+// valid UTF-8 character. Otherwise, it will return zero.
+static uint32_t
+one_utf8_to_utf32(const char* input, const size_t length) {
+    gunichar character = g_utf8_get_char_validated(input, length);
+    // Return 0 if there is more than one UTF-8 character:
+    if (g_unichar_to_utf8(character, NULL) != (gint)length)
+        return 0;
+    // Return 0 if the character is invalid.
+    if (character == (gunichar)-1 || character == (gunichar)-2)
+        return 0;
+    return character;
+}
+
+/** Get X11 keysym and a one-character representation from an Awesome keycode.
```

Comment:
Please use `/*` here rather than `/**`. If you want to expose this in the public API, use `awful.keyboard` (see other comment).`